packages/react-components/src/ConfigurationScreen.tsx

```diff
@@ -1,156 +1,40 @@
 import ToolIcon from './components/icons/ToolIcon';
-import ConfigurationInitialStep from './components/ConfigurationInitialStep';
-import ConfigurationUserPreferencesStep from './components/ConfigurationUserPreferencesStep';
-import ConfigurationCompletedStep from './components/ConfigurationCompletedStep';
-
-const TOAST_API_KEY_ERROR = 'api_key_error';
-const TOAST_TENANT_SUCCESS = 'tenant_success';
+import './global.css';
+import { ConfigurationStep } from './models/flare';
+import { createFlareIndex, fetchApiKey, redirectToHomepage } from './utils/setupConfiguration';
 
 const ConfigurationScreen: FC<{ theme: string }> = ({ theme }) => {
+    const [configurationStep, setConfigurationStep] = useState(ConfigurationStep.Initial);
     const [apiKey, setApiKey] = useState('');
-    const [tenantId, setTenantId] = useState(-1);
-    const [indexName, setIndexName] = useState(appName);
-    const [errorMessage, setErrorMessage] = useState('');
-    const [tenants, setUserTenants] = useState<Tenant[]>([]);
-    const [isIngestingMetadataOnly, setIsIngestingMetadataOnly] = useState(false);
-    const [isLoading, setIsLoading] = useState(false);
-    const [isCompleted, setIsCompleted] = useState(false);
-    const [indexNames, setIndexNames] = useState<string[]>([]);
 
     toastManager.setTheme(theme);
 
-    function reset(): void {
-        setApiKey('');
-        setTenantId(-1);
-        setUserTenants([]);
-        setIsLoading(false);
-        setIsCompleted(false);
-    }
-
-    function getCurrentConfigurationStep(): ConfigurationSteps {
-        if (tenants.length === 0) {
-            return ConfigurationSteps.Initial;
-        }
-        if (!isCompleted) {
-            return ConfigurationSteps.UserPreferences;
-        }
-
-        return ConfigurationSteps.Completed;
-    }
-
-    const handleApiKeyChange = (e): void => setApiKey(e.target.value);
-    const handleTenantIdChange = (e): void => setTenantId(parseInt(e.target.value, 10));
-    const handleIndexNameChange = (e): void => setIndexName(e.target.value);
-    const handleIsIngestingMetadataChange = (e): void => {
-        setIsIngestingMetadataOnly(e.target.checked);
-    };
-
     const handleBackButton = (): void => {
-        const currentConfigurationStep = getCurrentConfigurationStep();
-        if (currentConfigurationStep === ConfigurationSteps.Initial) {
+        if (configurationStep === ConfigurationStep.Initial) {
             redirectToHomepage();
-        } else if (currentConfigurationStep === ConfigurationSteps.UserPreferences) {
-            setUserTenants([]);
-        } else if (currentConfigurationStep === ConfigurationSteps.Completed) {
-            reset();
+        } else if (configurationStep === ConfigurationStep.UserPreferences) {
+            setConfigurationStep(ConfigurationStep.Initial);
+        } else if (configurationStep === ConfigurationStep.Completed) {
+            setConfigurationStep(ConfigurationStep.Initial);
```

Comment:
I feel like this is a good candidate for a switch. Would clean it up a bit and default should either be the for Initial and `redirectToHomepage()` or thow an unreachable error.

Reply:
Changed it to a switch and now throwing an error when a new step is added without implementing the back button action!